### PR TITLE
Implementation of Dependabot Alerts

### PR DESCRIPTION
**Pull request by Irma12**
This PR introduces Dependabot alerts in security insights plugin. Initial version will contain separate tab for dependabot alerts

Screenshot 2021-07-14 at 17 03 30

The tab contains the table displaying all the data in https://github.com/RoadieHQ/backstage-plugin-security-insights/blob/dependabot-integration/src/components/DependabotAlertsTable/DependabotAlertsTable.tsx

Screenshot 2021-07-14 at 17 03 40

### CODE REVIEW

File: src/components/DependabotAlertsTable/DependabotAlertsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+            <Box mr={1} />
+            <Typography variant="h6">{projectName}</Typography>
+          </Box>
+          <Paper>
+            <Box position="absolute" right={350} top={20}>
+              <ButtonGroup color="primary" aria-label="text primary button group">
+                <Button
+                  color={insightsStatusFilter === 'all' ? 'primary' : 'default'}
+                  onClick={() => {
+                    setInsightsStatusFilter(
```
Comment (Xantier) [inline src/components/DependabotAlertsTable/DependabotAlertsTable.tsx:121]: You could pull these three onClick callbacks into a method that takes the string and a filter function in as arguments so there's a bit less duplication here

**Comment by dtuite** [comment]
@Irma12 what's the first screenshot showing? Is that "Security insights" card part of this PR?

I think you could drop the "Details" column and instead turn the package name into a link. This pattern is used in a few places if I remember correctly, here's an example from the sentry plugin.

Screenshot 2021-07-15 at 10 22 25

**Comment by Irma12** [comment]
@dtuite Security Insights table and widget haven't been tackled with the PR, only Dependabot integration. I wanted to show we have a separate tab for Dependabot integration. Regrading the name link, good suggestion, will do that :)

**Review by Xantier — APPROVED** [review]
LGTM, could you check that the date format matches what is used elsewhere in the app as well for the table records

**Review by dtuite — APPROVED** [review]

**Comment by Irma12** [comment]
> LGTM, could you check that the date format matches what is used elsewhere in the app as well for the table records

@Xantier We usually represent how many days have passed from the specific date but I figured out date itself is also a good way. You think days count is better way to represent this?

**Comment by Xantier** [comment]
> @Xantier We usually represent how many days have passed from the specific date but I figured out date itself is also a good way. You think days count is better way to represent this?

In the screenshots we are displaying ISO8601 formatted date which is clearly the best format to use for dates and times anywhere ever. Unfortunately none of the countries in the world actually display their dates like that... :/

Cost insights uses 
```
export const dateFormatter = new Intl.DateTimeFormat('en-US', {
  timeZone: 'UTC',
  day: 'numeric',
  month: 'short',
});
``` 
Which is just silly american way of putting months before days so that might not be good. 

Elsewhere in Backstage Luxon is in use so maybe we can use that: https://moment.github.io/luxon/index.html#/formatting 
```
dt.toLocaleString(); //=> '4/20/2017'
```
might be enough for this iteration. It doesn't really matter, as long as it's consistent.

**Comment by Irma12** [comment]
@Xantier Good point, I will update it, thank you for bringing my attention on to this :)

**Comment by dtuite** [comment]
@Irma12 @Xantier [Luxon](https://github.com/backstage/backstage/issues/3401) is the blessed date library in Backstage so we should certainly look to use that where possible.

**Comment by Irma12** [comment]
@dtuite @Xantier Updated (together with screenshot in docs, which we will be displaying in Readme)

**Review by Xantier — APPROVED** [review]

**Pull request merged**